src/components/Input/Input.tsx

```diff
@@ -14,14 +15,35 @@ export type InputProps = (
 const Input = React.forwardRef<
   HTMLTextAreaElement | HTMLInputElement,
   InputProps
->(({ error = false, multiline = false, resize = "both", ...rest }, ref) => (
-  <Styled.Input
-    {...rest}
-    ref={ref as any}
-    as={(multiline ? "textarea" : "input") as any}
-    isError={error}
-    resize={resize}
-  />
-));
+>(({ error = false, multiline = false, resize = "both", ...rest }, ref) => {
+  if (rest.type === "number") {
+    const onWheel = (
+      event: React.WheelEvent<HTMLInputElement> &
+        React.WheelEvent<HTMLTextAreaElement>,
+    ) => {
+      event.currentTarget.blur();
+      if (rest.onWheel) rest.onWheel(event);
+    };
+    return (
+      <Styled.Input
+        {...rest}
+        ref={ref as any}
+        as={(multiline ? "textarea" : "input") as any}
+        isError={error}
+        resize={resize}
+        onWheel={onWheel}
+      />
+    );
+  }
+  return (
+    <Styled.Input
+      {...rest}
+      ref={ref as any}
+      as={(multiline ? "textarea" : "input") as any}
+      isError={error}
+      resize={resize}
+    />
```

Comment:
コンポーネントだし分けると行数増えるから、これでも良いかなぁと思ったけどどうだろう（試してないからわからないけど

```suggestion
    <Styled.Input
      {...rest}
      ref={ref as any}
      as={(multiline ? "textarea" : "input") as any}
      isError={error}
      resize={resize}
      onWheel={createChainedFunction(type === "number" ? handleWheel : null, rest.onWheel))}
      type={type === "number" ? "text" : type}
    />
```

Reply:
なるほど、createChainedFunction の存在を認知していませんでした、良さそうですね。